aspnetcore/blazor/performance/webassembly-browser-developer-tools-diagnostics.md

```diff
@@ -0,0 +1,120 @@
+`<WasmNativeStrip>` | `true` | `false` | Controls stripping the native executable.
+`<WasmNativeDebugSymbols>` | `true` | `true` | Controls building with native debug symbols.
+
+Enabling profilers has negative size and performance impact, so don't publish an app for production with profilers enabled. In the following example, a condition is set on a property group section that only enables profiling when the app is built with `/p:BlazorSampleProfilingEnabled=true` (.NET CLI) or `<BlazorSampleProfilingEnabled>true</BlazorSampleProfilingEnabled>` in a Visual Studio publish profile, where "`BlazorSampleProfilingEnabled`" is a custom symbol name that you choose and doesn't conflict with other symbol names.
+
+In the app's project file (`.csproj`):
+
+```xml
+<PropertyGroup Condition="'$(BlazorSampleProfilingEnabled)' == 'true'">
+  <WasmProfilers>browser;</WasmProfilers>
+  <WasmNativeStrip>false</WasmNativeStrip>
+  <WasmNativeDebugSymbols>true</WasmNativeDebugSymbols>
+</PropertyGroup>
+```
+
+Alternatively, enable features when the app is built with the .NET CLI. The following options passed to the `dotnet build` command mirror the preceding MS Build property configuration:
+
+```dotnetcli
+/p:WasmProfilers=browser /p:WasmNativeStrip=false /p:WasmNativeDebugSymbols=true
+```
+
+Setting WebAssembly profilers with `<WasmProfilers>browser;</WasmProfilers>` doesn't require AOT.
+
+The browser developer tools profiler can be used with AOT (`<RunAOTCompilation>`/`<RunAOTCompilationAfterBuild>` set to `true`) and without WebAssembly profilers (`<WasmProfilers>browser;</WasmProfilers>` removed from the preceding property group).
+
+To see AOT method names in the developer tools console, install [DWARF chrome extension](https://chromewebstore.google.com/detail/cc++-devtools-support-dwa/pdcpmagijalfljmkmjngeonclgbbannb).
+
+## Set the sample interval
+
+Configure the sample interval in the app's project file. In the following example, the `{INTERVAL}` placeholder represents the time in milliseconds. The default setting if `sampleIntervalMs` isn't specified is 10 ms.
+
+```xml
+<PropertyGroup>
+  <WasmProfilers>browser:sampleIntervalMs={INTERVAL};</WasmProfilers>
+</PropertyGroup>
+```
+
+## Call specification (:::no-loc text="callspec":::)
+
+If you want to filter profiled methods, you can use call specification (:::no-loc text="callspec":::). For more information, see [Trace MonoVM profiler events during startup](https://github.com/dotnet/runtime/blob/main/docs/design/mono/diagnostics-tracing.md#trace-monovm-profiler-events-during-startup).
+
+Add `callspec` to the `browser` WebAssembly profiler in the `<WasmProfilers>` element. In the following example, the `{APP NAMESPACE}` placeholder is the app's namespace:
+
+```xml
+<WasmProfilers>browser:callspec=N:{APP NAMESPACE};</WasmProfilers>
+```
+
+## .NET Core Diagnostics Client Library example
+
+Parse and validate NetTrace (`.nettrace`) messages using the .NET Core Diagnostics Client Library:
```

Comment:
`Parse and validate NetTrace` is EP related, doesn't belon to this page. Drop lines up to 100